src/frequenz/sdk/timeseries/_periodic_feature_extractor.py

```diff
@@ -0,0 +1,419 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""
+A feature extractor for historical timeseries data.
+
+It creates a profile from periodically occurring windows in a buffer of
+historical data.
+
+The profile is created out of all windows that are fully contained in the
+underlying buffer, e.g. the moving window, with the same start and end time
+modulo a fixed period.
+"""
+
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+from typing import List, Tuple
+
+import numpy as np
+from numpy.typing import NDArray
+
+from .._internal._math import is_close_to_zero
+from ._moving_window import MovingWindow
+from ._ringbuffer import OrderedRingBuffer
+
+_logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class RelativePositions:
+    """
+    Holds the relative positions of a window in a buffer.
+
+    When calculating a profile the user has to pass two datetime objects which
+    are determaning a window. It's allowed that this window is not contained in
+    the historical data buffer.
+    In order to calculate the profile the window is shifted to the first
+    position in the buffer where it occurrs first and represented by indexes
+    relative to the oldest sample in the buffer.
+    """
+
+    start: int
+    """The relative start position of the window."""
+    end: int
+    """The relative end position of the window."""
+    next: int
+    """The relative position of the next incoming sample sample."""
```

Comment:
```suggestion
    """The relative position of the next incoming sample."""
```

Reply:
That's intended! Having it twice makes a bigger impact :D.